Added /fly command and adjusted sleep requirement.

junerhobart opened the pull request:
### Changes

- Added `/fly` command Players can now fly within their own region or any region where they have build permissions.

- Adjusted bedtime percentage Reduced required sleep percentage from **50%** to **20%**.

Commit:
Added /fly command and adjusted sleep requirement.

### Changes

- Added `/fly` command
  Players can now fly within their own region or any region where they have build permissions.

- Adjusted bedtime percentage
  Reduced required sleep percentage from **50%** to **20%**.

Commit:
Updated the webpage to reflect my previous changes

Updated the webpage to reflect my previous changes being /fly in regions and 50% sleep requirement to 20%

Commit:
Added clearer documentation for graylist

Commit:
Merge branch 'oddlama:main' into main

Commit:
Fixed translation issue

Commit:
Merge branch 'main' of https://github.com/junerhobart/vane

Commit:
Fixed a mountian of bugs

oddlama commented on documentation/content/vane-admin/graylist.md:
```diff
@@ -18,6 +18,18 @@ you can assign a new player to the `user` group by executing `perm add player_na
 
+### Quick start
+
+- Enable in `vane-admin` config: set `world_protection.enabled: true`, then run `/vane reload`.
```
Let's only state the configuration that must be changed, and not mention `/vane reload` in this context. I don't want to proliferate any usage of server or plugin config reload (even though the vane-internal command should do the right thing most of the time). Since server reloads are always the reason for subtle bugs in many plugins I think it is better to always recommend restarting the server.

oddlama commented on vane-regions/src/main/java/org/oddlama/vane/regions/Regions.java:
```diff
@@ -706,7 +799,7 @@ public void load_persistent_data(final World world) {
             .getKeys()
             .stream()
             .filter(key -> key.toString().startsWith(storage_region_prefix))
-            .map(key -> StringUtils.removeStart(key.toString(), storage_region_prefix))
```
Why change this? Looks equivalent

junerhobart committed:
Merge branch 'oddlama:main' into main

Commit:
address PR review feedback

- revert sleep threshold default back to 0.5
- restore org.json relocation in build.gradle.kts
- import HashSet directly instead of fully-qualified
- fall damage protection now always applies, not just when high up
- simplify fall damage check using remove()
- use StringUtils.removeStart instead of substring
- drop /vane reload from graylist docs, just restart
- fix missing leading slash on perm add commands in graylist
- move flight in regions docs to regions.md source file

Sylfare commented on documentation/content/vane-admin/graylist.md:
```diff
@@ -18,6 +18,18 @@ you can assign a new player to the `user` group by executing `perm add player_na
 
+### Quick start
+
+- Enable in `vane-admin` config: set `world_protection.enabled: true`, then restart the server.
```
Does `/vane reload` works here, or do you really have to restart the server?

Sylfare commented on docs/index.html:
```diff
@@ -259,7 +259,7 @@ <h2 id="feature-vane-bedtime--night_to_day">
 <div id="body-feature-vane-bedtime--night_to_day" class="hidden">
 	<div class="p-5 border border-b-0 border-gray-200 dark:border-gray-700 dark:bg-gray-900">
-		<div class="prose prose-slate prose-img:rounded-xl prose-img:w-1/3 prose-img:sm:m-4 prose-img:w-full prose-img:sm:w-1/2 prose-img:lg:w-2/5 prose-img:min-w-32 prose-img:max-w-full prose-img:float-right prose-img:clear-right prose-h1:clear-right prose-h2:clear-right prose-img:float-right dark:prose-invert max-w-none"><p>Once at least 50% of all players in the world are asleep, the night is skippe ...
+		<div class="prose prose-slate prose-img:rounded-xl prose-img:w-1/3 prose-img:sm:m-4 prose-img:w-full prose-img:sm:w-1/2 prose-img:lg:w-2/5 prose-img:min-w-32 prose-img:max-w-full prose-img:float-right prose-img:clear-right prose-h1:clear-right prose-h2:clear-right prose-img:float-right dark:prose-invert max-w-none"><p>Once at least 20% of all players in the world are asleep, the night is skippe ...
```
The limit is back to 50%, you should regenerate the docs

Sylfare commented on vane-regions/src/main/java/org/oddlama/vane/regions/RegionFlyManager.java:
```diff
@@ -0,0 +1,247 @@
+    }
+    
+    private boolean can_fly_in_region(final Player player, final Region region) {
+        if (region == null) {
+            return false;
+        }
```
I have a doubt about this: does it allows players who can fly with some other ways, like Essentials' `/fly` command?

Sylfare commented on vane-regions/src/main/java/org/oddlama/vane/regions/RegionFlyManager.java:
```diff
@@ -0,0 +1,247 @@
+            return;
+        }
+        
+        final var player = (Player) event.getEntity();
```
You can avoid the cast on L211 by adding the variable name after `instanceof`

```java
if (!(event.getEntity() instanceof Player player)) {
    return;
}
      
// Only handle fall damage
if (event.getCause() != EntityDamageEvent.DamageCause.FALL) {
    return;
}

final var player_id = player.getUniqueId();
```

Sylfare commented on vane-trifles/build.gradle.kts:
```diff
@@ -13,8 +13,8 @@ dependencies {
 }
 
 tasks {
-	shadowJar {
-		configurations = listOf()
-		relocate("org.json", "org.oddlama.vane.external.json")
-	}
```
I don't know enough about shadowing, but I'm not sure this is a good idea to remove it

Sylfare reviewed (commented):
Other lang files are not updated yet

Commit:
more review fixes

- fix bedtime docs still showing 20% instead of 50%
- restore org.json relocation in vane-trifles
- use instanceof pattern matching in fall damage handler
- don't manage flight for players who already have it from elsewhere (Essentials etc)
- add fly/visualize lang keys to all non-English lang files, bump to version 4
